### DISCUSSION TITLE
"Renovate has not found any CVEs on osv.dev." when this isn't true

### DISCUSSION THREAD

**Discussion by nitrocode** [renovatebot/renovate#31089] [Request Help] [answered]
### What would you like help with?

None

### How are you running Renovate?

None

### Please tell us more about your question or problem

```json5
{
  dependencyDashboardOSVVulnerabilitySummary: "all",
}
```

I see this show up in Snyk https://security.snyk.io/vuln/SNYK-JAVA-ORGAPACHETOMCATEMBED-7430175 and CVE https://www.cve.org/CVERecord?id=CVE-2024-34750

I see the same vulnerability show up in

https://osv.dev/list?q=CVE-2024-34750&ecosystem=
https://osv.dev/vulnerability/GHSA-wm9w-rjj3-j356

In renovate, I see this dependency detected org.springframework.boot 2.7.18 which if upgraded to springboot 3.x, it will fix the issue.

From Snyk UI

    Introduced through
    org.springframework.boot:spring-boot-starter-web@2.7.18
    Fixed in
    org.apache.tomcat.embed:tomcat-embed-core@9.0.90, @10.1.25, @11.0.0-M21

---

How come renovate's issue dashboard shows the message "Renovate has not found any CVEs on osv.dev."

How does renovate search the osv database?

Is there a way to alter what we send to osv?

Can we use a different database?

### Logs (if relevant)

N/A

**Comment by RahulGautamSingh** [comment]
You will only get OSV-based vulnerability alerts for direct dependencies.

**Reply by nitrocode** [reply]
Oh I see. I was very confused when I saw there were no vulnerabilities.

Could we

- Change the text in the issue dashboard to show `Renovate has not found any CVEs on osv.dev for direct dependencies. Indirect dependencies may have vulnerabilities.` or similar
- Change the parameter documentation for `dependencyDashboardOSVVulnerabilitySummary` to include that this will only show alerts for direct dependencies ?

Thanks!

**Reply by RahulGautamSingh** [reply, marked as answer]
Mentioning it in the docs sounds good. In the Dashboard it just seems like extra/redundant info.

**Reply by nitrocode** [reply]
For managers and developers who are unfamiliar with the nuances of this configuration, will only see `Renovate has not found any CVEs on osv.dev` and assume there are no vulnerabilities. It also gives them plausible deniability.

While I disagree that the information is redundant (I'd argue it's more accurate albeit longer :) ), we can provide the additional context using [dependencyDashboardHeader](https://docs.renovatebot.com/configuration-options/#dependencydashboardheader).

Thanks again for the support and the consideration!
